src/backend/distributed/commands/table.c

```diff
@@ -939,6 +939,32 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					Relation rel = RelationIdGetRelation(leftRelationId);
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
```

Comment:
It would be more readable to specify that we are passing a true boolean value for this index being a primary key. Something like
```C
bool primary = true;
bool isconstraint = true;
ChooseIndexName(relname, namespace, null, null, primary, isconstraint);
```